### ISSUE TITLE
Reduce duplicated indicator source code

### ISSUE THREAD

**Issue by Zeitsperre** [Ouranosinc/xclim#744]
With the push to synchronize our code base with the `clix-meta` project, it may be worthwhile to start deprecating some indicators we offer currently. One example of duplicated indicators is:

- `xclim.indices.daily_temperature_range_variability` and 
  - https://github.com/Ouranosinc/xclim/blob/e425aac1ecf56f6624610657ad1946ea82e1802c/xclim/indices/_multivariate.py#L357 
- `xclim.indices.generic.interday_diurnal_temperature_range` 
  - https://github.com/Ouranosinc/xclim/blob/e425aac1ecf56f6624610657ad1946ea82e1802c/xclim/indices/generic.py#L602

This obviously entails breaking changes, so identifying all affected indicators and gradually deprecating them is the safest way forward.

**Comment by Zeitsperre** [comment]
Here's another such example:

* https://github.com/Ouranosinc/xclim/blob/798c036a1022e8e1ad70cc98bba3a69144719aa4/xclim/indices/_multivariate.py#L564
* https://github.com/Ouranosinc/xclim/blob/798c036a1022e8e1ad70cc98bba3a69144719aa4/xclim/indices/generic.py#L657

I suppose in this case, we could offer both, seeing as the `_multrivariate` version has units checking, but we could also just port this units checking directly to `generic`, no?

**Comment by aulemahal** [comment]
Indeed. This example exists because it's part of the clix-meta functions, and for this reason the `generic` one is not decorated with `declare_units`. We _could_ port the unit checking to  `generic` but that might break the automatic build of the `cf`module and it would be incoherent with the other clix-meta functions (which are not units-ready)...
I can accept this duplication, but we could also make the `_multivariate` version directly call the `generic` one, with added unit checking and a better signature.

**Comment by Zeitsperre** [comment]
Interesting. This is certainly a bit confusing (I wasn't sure what the difference was at first glance). It might be good to add a `Warning` to these, or maybe better flesh out their methodologies? Perhaps this is a case for a `method` switch?

**Comment by aulemahal** [comment]
Well, the one in `generic` is defined by clix-meta, so I'd be reluctant to change its name or signature. BUT, the docstring needs a fix, because it's currently wrong.

However, implementing this other version in the `_multivariate` one could be interesting, indeed using a `method` switch.

**Comment by aulemahal** [comment]
Hum, I just went and read the official definition and it seems it is xclim that is wrong. The  `generic` function should do the same as the `_multivariate` one! I wrongly implemented it, or so it seems!

**Comment by Zeitsperre** [comment]
Ouch. That happens. Open an issue about it for reference?
